### PR TITLE
Admin notice for OPCache clearing

### PR DESCRIPTION
**Pull request by crystinutzaa**
Added admin notices for OPCache clearing

OPCache clearing - Issue demo - https://jmp.sh/JZiRHTc

Based on:
3.4.1 WP Rocket Minor Version doc

### CODE REVIEW

File: inc/admin/ui/notices.php [hunk 1]
```diff
@@ -627,6 +627,36 @@ function rocket_cloudflare_purge_result() {
...
 add_action( 'admin_notices', 'rocket_cloudflare_purge_result' );
 
+/**
+ * This notice is displayed after purging OPcache
+ *
+ * @since 3.4.1
+ * @author Soponar Cristina
+ */
+function rocket_opcache_purge_result() {
+	global $current_user;
```
Comment (Tabrisrp) [inline inc/admin/ui/notices.php]:
```suggestion
```

File: inc/admin/ui/notices.php [hunk 2]
```diff
@@ -627,6 +627,36 @@ function rocket_cloudflare_purge_result() {
...
+	if ( ! current_user_can( 'rocket_purge_opcache' ) ) {
+		return;
+	}
+
+	if ( ! is_admin() ) {
+		return;
+	}
+
+	$notice = get_transient( $current_user->ID . '_opcache_purge_result' );
```
Comment (Tabrisrp) [inline inc/admin/ui/notices.php]:
```suggestion
        $user_id = get_current_user_id();
	$notice = get_transient( $user_id . '_opcache_purge_result' );
```

File: inc/admin/ui/notices.php [hunk 3]
```diff
@@ -627,6 +627,36 @@ function rocket_cloudflare_purge_result() {
...
+	if ( ! $notice ) {
+		return;
+	}
+
+	delete_transient( $current_user->ID . '_opcache_purge_result' );
```
Comment (Tabrisrp) [inline inc/admin/ui/notices.php]:
```suggestion
	delete_transient( $user_id . '_opcache_purge_result' );
```

File: inc/common/purge.php [hunk 4]
```diff
@@ -436,7 +436,21 @@ function do_admin_post_rocket_purge_opcache() {
...
+	if ( ! $reset_opcache ) {
+		$op_purge_result = array(
+			'result'  => 'error',
+			'message' => __( 'OPcache purge failed.', 'rocket' ),
+		);
+	} else {
+		$op_purge_result = array(
+			'result'  => 'success',
+			'message' => __( 'OPcache successfully purged', 'rocket' ),
+		);
```
Comment (Tabrisrp) [inline inc/common/purge.php]: We usually use the shorthand syntax for arrays, `[]`instead of `array()`

File: inc/common/purge.php [hunk 5]
```diff
@@ -436,7 +436,21 @@ function do_admin_post_rocket_purge_opcache() {
...
+	}
+
+	set_transient( $GLOBALS['current_user']->ID . '_opcache_purge_result', $op_purge_result );
```
Comment (Tabrisrp) [inline inc/common/purge.php]:
```suggestion
	set_transient( get_current_user_id() . '_opcache_purge_result', $op_purge_result );
```

File: inc/common/purge.php (hunk 4 above)
Comment (crystinutzaa) [reply inc/common/purge.php]: Replaced all array() with the shorthand [] syntax in the files:
`/inc/common/purge.php`
`/inc/admin/ui/notices.php`

File: inc/common/purge.php (hunk 5 above)
Comment (crystinutzaa) [reply inc/common/purge.php]: Replaced all `$GLOBALS['current_user']->ID` with `get_current_user_id()` in the files:
`/inc/common/purge.php`
`/inc/admin/ui/notices.php`

**Pull request merged**